domain/application/state/application.go

```diff
@@ -62,8 +62,10 @@ func (st *State) GetModelType(ctx context.Context) (coremodel.ModelType, error)
 
 // CreateApplication creates an application, returning an error satisfying
 // [applicationerrors.ApplicationAlreadyExists] if the application already exists.
-// If returns as error satisfying [applicationerrors.CharmNotFound] if the charm
-// for the application is not found.
+// It returns as error satisfying [applicationerrors.CharmNotFound] if the charm
+// for the application is not found. It may also return an error satisfying
+// [applicationerrors.InvalidResourceArgs] if there is a mismatch between charm
+// resources and resource arguments.
```

Comment:
todo: please update, there is no validation of the resources happening in this method.